### PR TITLE
Migrate to vtex.css-handles@1.x

### PR DESCRIPTION
**Pull request by victorhmp**
#### What problem is this solving?

We've recently released a new major version of `vtex.css-handles`, in which we implemented our vision for what CSS customization on Store Framework should look like from now on.

This PR makes the necessary changes in this app's code to accommodate the API changes introduced by `vtex.css-handles@1.x`. It also adds a new CSS handle and adds a `classes` prop to every component exported by the app.

Should close https://github.com/vtex-apps/store-discussion/issues/493:

Screen Shot 2020-12-10 at 11 25 36

#### How to test it?

There should be no visible difference at [Workspace](https://victormiranda--storecomponents.myvtex.com/).

#### Related to / Depends on

This is related to: https://github.com/vtex-apps/css-handles/pull/22.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -136,11 +138,12 @@ The available `menu-item` block props are as follows:
...
+| `highlight`         | `boolean` | Whether the item has highlight                                            | `undefined`           |
 | `iconPosition`         | `String` | Icon position relative to the menu item text. Either to the `left` or `right`                                           | `left`          |
-| `iconProps`         | `IconProps` | Icon props                                           | N/A           |
-| `itemProps`         | `CategoryItem` or `CustomItem` | Item props                                           | N/A           |
+| `iconProps`         | `IconProps` | Icon props                                           | `undefined`           |
+| `itemProps`         | `CategoryItem` or `CustomItem` | Item props                                           | `undefined`           |
+| `classes`         | `CustomCSSClasses` | Used to override default CSS handles. To better understand how this prop works, we recommend reading about it [here](https://github.com/vtex-apps/css-handles#usecustomclasses). Note that this is only useful if you're using this block as a React component.                                      | `undefined`           |
```
Comment (kaisermann) [inline docs/README.md:146]:
```suggestion
| `classes`         | `CustomCSSClasses` | Used to override default CSS handles. To better understand how this prop works, we recommend reading about it [here](https://github.com/vtex-apps/css-handles#usecustomclasses). Note that this is only useful if you're importing this block as a React component.                                      | `undefined`           |
```

File: react/SubmenuAccordion.tsx [hunk 2]
```diff
@@ -10,19 +9,19 @@ const CSS_HANDLES = ['submenuAccordion'] as const
...
+  classes,
   children,
 }) => {
-  const handles = useCssHandles(CSS_HANDLES)
+  const { handles } = useCssHandles(CSS_HANDLES, { classes })
 
   return (
     <Collapsible open={isOpen}>
-      <section
-        className={classNames(handles.submenuAccordion, 'w-100 flex pl4 flex')}
```
Comment (kaisermann) [inline react/SubmenuAccordion.tsx:24]: Caraca que ódio AHAHAHAHAH

**Review by kaisermann — APPROVED** [review]

**Review by vitorflg — APPROVED** [review]

**Review by klzns — APPROVED** [review]

File: react/Column.tsx [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
+import { useCssHandles, CssHandlesTypes } from 'vtex.css-handles'
 
-const Column: FunctionComponent<Props> = props => {
-  return <div>{props.children}</div>
-}
+const CSS_HANDLES = ['submenuColumn'] as const
 
 interface Props {
-  children: React.ReactChildren
+  classes: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```
Comment (icazevedo) [inline react/Column.tsx:7]:
```suggestion
  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```

File: react/Menu.tsx [hunk 4]
```diff
@@ -12,6 +12,29 @@ import MenuItem, { MenuItemSchema } from './MenuItem'
...
+interface MenuSchema {
+  orientation?: 'vertical' | 'horizontal'
+  categoryId?: number
+  textType?: Typography
+  title?: MenuItemSchema
+  additionalDef?: string
+  blockClass?: string
+  items?: MenuItemSchema[]
+  experimentalOptimizeRendering?: boolean
+  classes: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```
Comment (icazevedo) [inline react/Menu.tsx:24]:
```suggestion
  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```

File: react/MenuItem.tsx [hunk 5]
```diff
@@ -21,7 +21,19 @@ import MenuContext from './components/MenuContext'
...
+export interface MenuItemSchema {
+  id: string
+  type: 'category' | 'custom'
+  iconProps: IconProps
+  iconPosition: 'left' | 'right'
+  highlight: boolean
+  itemProps: CategoryItemSchema | CustomItemSchema
+  blockClass?: string
+  experimentalOptimizeRendering?: boolean
+  classes: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```
Comment (icazevedo) [inline react/MenuItem.tsx:35]:
```suggestion
  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```

File: react/Submenu.tsx [hunk 6]
```diff
@@ -1,10 +1,19 @@
...
 
 const CSS_HANDLES = ['submenu', 'submenuWrapper', 'submenuContainer'] as const
 
+export interface SubmenuProps extends SubmenuSchema {
+  isOpen: boolean
+  orientation: Orientation
+  paddingTop: number | string
+  paddingBottom: number | string
+  blockClass?: string
+  classes: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```
Comment (icazevedo) [inline react/Submenu.tsx:14]:
```suggestion
  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```

File: react/SubmenuAccordion.tsx [hunk 7]
```diff
@@ -10,19 +9,19 @@ const CSS_HANDLES = ['submenuAccordion'] as const
 interface Props {
   isOpen: boolean
   blockClass?: string
+  classes: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```
Comment (icazevedo) [inline react/SubmenuAccordion.tsx:12]:
```suggestion
  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```

File: react/index.tsx [hunk 8]
```diff
@@ -73,10 +76,11 @@ const getValidPage = (page?: string | null): string => {
  * Shows a menu bar with links.
  */
 const MenuLink: StorefrontFunctionComponent<Props> = ({ links = [] }) => {
-  const handles = useCssHandles(CSS_HANDLES)
+  const { handles } = useCssHandles(CSS_HANDLES)
```
Comment (icazevedo) [inline react/index.tsx:79]: Is there a reason why this component doesn't have `classes`?

File: react/components/StyledLink.tsx [hunk 9]
```diff
@@ -24,7 +24,7 @@ const defaultTypography: Record<number, string> = {
 const StyledLink: FunctionComponent<StyledLinkProps> = props => {
   const level = useContext(LevelContext)
   const { orientation, hasTitle } = useContext(MenuContext)
-  const handles = useCssHandles(CSS_HANDLES)
+  const { handles, withModifiers } = useCssHandles(CSS_HANDLES)
```
Comment (icazevedo) [inline react/components/StyledLink.tsx:27]: Is there a reason why this component doesn't have classes?

**Review by icazevedo — CHANGES_REQUESTED** [review]

File: react/Submenu.tsx [hunk 10]
```diff
@@ -44,7 +53,7 @@ const Submenu: StorefrontFunctionComponent<SubmenuProps> = ({
   paddingTop = 4,
   paddingBottom = 4,
 }) => {
-  const handles = useCssHandles(CSS_HANDLES)
+  const { handles, withModifiers } = useCssHandles(CSS_HANDLES)
```
Comment (icazevedo) [inline react/Submenu.tsx:56]: I think you forgot to use the `classes` prop here

**Review by icazevedo — CHANGES_REQUESTED** [review]

**Review by guerreirobeatriz — APPROVED** [review]

File: react/index.tsx (hunk 8 above)
Comment (victorhmp) [reply react/index.tsx:79]: Yep, I forgot 😂

File: react/components/StyledLink.tsx (hunk 9 above)
Comment (victorhmp) [reply react/components/StyledLink.tsx:27]: Yep, because this component is not exported by the app

File: react/Submenu.tsx (hunk 10 above)
Comment (victorhmp) [reply react/Submenu.tsx:56]: Good catch!

**Review by icazevedo — APPROVED** [review]

**Pull request merged**